tests/constraints.in

```diff
@@ -0,0 +1,5 @@
+# Known limitations for indirect/transitive dependencies.
+resolvelib < 1.1.0
+sphinx == 5.3.0
```

Comment:
Why are we pinned to this specific sphinx version from ten months ago?

Reply:
That's just migrated from the original file, I recommend exploring this in a follow-up.

Reply:
The Sphinx version is pinned for the same reason as why antsibull-docs is pinned to a specific version: because it has a huge impact on the result. That doesn't mean it shouldn't be updated, but that updates should be done with enough testing.

Reply:
@felixfontein I suppose pins in the constraint file have the same function. The incoming constraints should probably guard the major version updates (although, that's probably unnecessary). But that's about it. With manual pip-tools runs, it's possible to skip updating sphinx even if it's not in the constraints files.

Though, I think you're mostly concerned about differences in the HTML output, so maybe a CI check could compare the generated output for the upgraded pin vs. a pre-PR commit.

Reply:
The pin doesn't come from my side, I only explainde why it is there. I would also pin everything to major versions (and use these pins for devel builds), and use the strict pip-compile result for stable builds.

Reply:
We pin sphinx until we formally test the next updated version. Since this goes to production (aka docs.ansible.com) I'd like to keep it pinned. My recommendation is that we keep versions 'as is' in this PR and once the new way of doing these test requirements has completed, we open an issue to update versions. That way we are formally testing new versions in a separate PR and not doing two things at once in this PR.